src/client/signatures.rs

```diff
@@ -70,8 +71,9 @@ fn epoch_seconds() -> u64 {
 ///    .build();
 /// # }
 /// ```
-pub trait Verification: Send {
-    fn fetch_certificate_chain(
+#[async_trait]
+pub trait Verification: Send + Sync {
+    async fn fetch_certificate_chain(
```

Comment:
@leplatrem should all the methods of this trait be `async`?

Reply:
> I believe `fetch_certificate_chain()` and thus `verify()` is enough.
> 
> I don't think serializing data and calling crypto primitives should be. If some day we find ourselves in a situation where they should, then we'll change it. What do you think?

@leplatrem I agree. We can cross that bridge once we see a problem with the current approach :-)

I'll fix the test and merge this :)